<launch>
  <arg name="test_mode" default="false" />
  <arg name="gui" default="true" unless="$(arg test_mode)" />
  <arg name="gui" default="false"    if="$(arg test_mode)" />
  <arg name="GPU" default="0" unless="$(arg test_mode)"  />
  <arg name="GPU" default="-1"    if="$(arg test_mode)" />
  <arg name="model_path" default="auto" />
  <arg name="CAMERA" default="zed" />
  <arg name="INPUT_IMAGE" default="$(arg CAMERA)/rgb/image_rect_color" />
  <arg name="INPUT_DEPTH_IMAGE" default="$(arg CAMERA)/depth/depth_registered" />
  <arg name="INPUT_CAMERA_INFO" default="$(arg CAMERA)/rgb/camera_info" />

  <node name="face_pose_estimation"
        pkg="jsk_perception" type="face_pose_estimation.py"
        output="screen">
    <remap from="~input" to="$(arg INPUT_IMAGE)"/>
    <remap from="~input/pose_2d" to="people_pose_estimation_2d/pose_2d" />
    <remap from="~input/pose" to="people_pose_estimation_2d/pose" />
    <rosparam subst_value="true">
      gpu: $(arg GPU)
    </rosparam>
    <param name="model_path" value="$(arg model_path)" unless="$(arg test_mode)" />
  </node>

  <node name="draw_rects"
        pkg="jsk_perception" type="draw_rects">
    <remap from="~input" to="$(arg INPUT_IMAGE)"/>
    <remap from="~input/rects" to="face_pose_estimation/output/rects" />
    <remap from="~input/class" to="face_pose_estimation/output/gender" />
    <rosparam>
      use_classification_result: true
    </rosparam>
  </node>

  <node name="people_poses_to_poses"
        pkg="jsk_recognition_msgs" type="people_pose_array_to_pose_array.py">
    <remap from="~input" to="people_pose_estimation_2d/pose" />
  </node>
ro
</launch>
